ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -37,10 +40,54 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = &signed_address_change.message;
+    let signature = &signed_address_change.signature;
+
+    if address_change.validator_index >= state.validators.len() {
+        return Err(invalid_operation_error(InvalidOperation::ValidatorIndex(
+            address_change.validator_index,
+        )))
+    }
+
+    let withdrawal_credentials_prefix =
+        state.validators[address_change.validator_index].withdrawal_credentials[0];
+
+    if withdrawal_credentials_prefix != BLS_WITHDRAWAL_PREFIX {
+        return Err(invalid_operation_error(InvalidOperation::WithdrawalCredentialsPrefix(
+            state.validators[address_change.validator_index].withdrawal_credentials[0],
+        )))
+    }
+
+    let public_key: &BlsPublicKey = &address_change.from_bls_public_key;
+
+    let domain = compute_domain(
+        DomainType::BlsToExecutionChange,
+        None,
+        Some(state.genesis_validators_root),
+        context,
+    )?;
+    let signed_data =
+        verify_signed_data(&mut address_change.clone(), signature, public_key, domain);
+    if signed_data.is_err() {
+        return Err(invalid_operation_error(InvalidOperation::ExecutionChange(
+            signed_address_change.signature.clone(),
+        )))
+    }
```

Comment:
the API here is that it only returns an error if the signature is invalid, and otherwise just `Ok(())`

so we can do this and even skip bothering w/ the specific error variant
```suggestion
    verify_signed_data(&mut address_change.clone(), signature, public_key, domain)?;
```